action.yml

```diff
@@ -16,5 +16,5 @@ runs:
     - name: filter
       run: |
         unset LD_PRELOAD
-        python3 "${{ github.action_path }}/filter_sarif.py" --input "${{ inputs.input }}" --output "${{ inputs.output }}" --split-lines -- "${{ inputs.patterns }}"
+        python3 "$GITHUB_ACTION_PATH/filter_sarif.py" --input "${{ inputs.input }}" --output "${{ inputs.output }}" --split-lines -- "${{ inputs.patterns }}"
```

Comment:
Sorry, I don't really understand.

There are two ways of accessing the action path within a composite action:

1. Via a context expression: `${{ github.action_path }}`
2. Via an env var: `$GITHUB_ACTION_PATH`

Approach (1) does not work properly inside a container. Approach (2) does.

By switching the action to use approach (2), it will will work inside containers. It currently doesn't work inside containers.

Reply:
For reference I ran into similar problems with the `${{ runner.temp }}` variable and raised an issue https://github.com/actions/runner/issues/2498 . On that issue there is some explanation on why things behave the way they do. The Actions team is aware of the problem but it is a bit tricky to fix. The recommendation is indeed to use the environment variable.